lib/ast/AST.cpp

```diff
@@ -1122,6 +1122,19 @@ sptr<kore_pattern> kore_composite_pattern::unflatten_and_or() {
   return result;
 }
 
+sptr<kore_pattern> kore_composite_pattern::strip_injections() {
+  if (constructor_->get_name() == "inj" && arguments_.size() == 1) {
+    return arguments_[0]->strip_injections();
+  }
```

Comment:
Shouldn't we still check if the `constructor_` has the `inj`  name before constructing the result object?

Reply:
This implementation is recursive; if we go past this `if` block then we know all injections have been stripped.